Classes/core/entity/tower/special_magician_plus.cpp

```diff
@@ -7,52 +7,51 @@ namespace towerdefence {
                 : Tower(id, clk), release_skill_(clk.with_period_sec(20)) {}
         void SpecialMagicianPlus::on_tick(GridRef g) {
             Tower::on_tick(g);
+            this->timeouts_.on_tick(g.clock(), *this, g);
             auto &clk = g.clock();
 
             auto status = this->status();
-            // do normal attack
+
             if (clk.is_triggered(this->attack_)){
                 auto grids = g.with_radius(status.attack_radius_, linf_dis);
                 auto enemy_grid = get_enemy_grid(*this,grids);
                 if(enemy_grid!=grids.end()){
-                    this->add_buff({this->id, Buff::DEFAULT},
+                    if(skill){
+                        this->add_buff({this->id, Buff::DEFAULT},
+                                       Buff::attack(1.50 * layer));
+                        SpecialMagicianPlus &self = *this;
+                        auto buffs = self.get_all_buff();
+                        enemy_grid->with_nearest_enemy([&self, buffs](Enemy &target_enemy) {

```

Comment:
Actually `this` is ok here since `with_nearest_enemy` will not store the lambda and call it immediately